RiotSharp/RiotApi.cs

```diff
@@ -1046,5 +1053,136 @@ public FeaturedGames GetFeaturedGames(Region region)
                 region);
             return (await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<FeaturedGames>(json)));
         }
+
+        /// <summary>
+        /// Gets a champion mastery by summoner ID synchronously.
+        /// </summary>
+        /// <param name="platform">Region where to retrieve the data.</param>
+        /// <param name="summonerId">ID of the summoner for which to retrieve champion mastery.</param>
+        /// <param name="championId">ID of the champion for which to retrieve mastery.</param>
+        /// <returns>Champion mastery for summoner ID and champion ID.</returns>
+        public ChampionMastery GetChampionMastery(Platform platform, long summonerId, long championId)
+        {
+            var rootUrl = string.Format(ChampionMasteryRootUrl, platform, summonerId);
+            var additionalUrl = string.Format(ChampionMasteryByChampionId, championId);
+
+            var json = requester.CreateGetRequest(rootUrl + additionalUrl, platform.ConvertToRegion());
+            return JsonConvert.DeserializeObject<ChampionMastery>(json);
+        }
+
+        /// <summary>
+        /// Gets a champion mastery by summoner ID asynchronously.
+        /// </summary>
+        /// <param name="platform">Region where to retrieve the data.</param>
+        /// <param name="summonerId">ID of the summoner for which to retrieve champion mastery.</param>
+        /// <param name="championId">ID of the champion for which to retrieve mastery.</param>
+        /// <returns>Champion mastery for summoner ID and champion ID.</returns>
+        public async Task<ChampionMastery> GetChampionMasteryAsync(Platform platform,
+            long summonerId, long championId)
+        {
+            var rootUrl = string.Format(ChampionMasteryRootUrl, platform, summonerId);
+            var additionalUrl = string.Format(ChampionMasteryByChampionId, championId);
+
+            var json = await requester.CreateGetRequestAsync(rootUrl + additionalUrl, platform.ConvertToRegion());
+            return (await Task.Factory.StartNew(() => JsonConvert.DeserializeObject<ChampionMastery>(json)));
+        }
+
+        /// <summary>
+        /// Gets all champions mastery by summoner ID synchronously.
+        /// </summary>
+        /// <param name="platform">Region where to retrieve the data.</param>
+        /// <param name="summonerId">ID of the summoner for which to retrieve champion mastery.</param>
+        /// <returns>All champions mastery for summoner ID.</returns>
+        public List<ChampionMastery> GetAllChampionsMastery(Platform platform, long summonerId)
```

Comment:
I think this should be renamed in `GetAllChampionMasteries` or `GetAllChampionMasteryEntries` so we know to expect a list, what do you think?